### PR TITLE
chore: Fix eslint rule @typescript-eslint/ban-types

### PR DESCRIPTION
**Pull request by scinos**
See #56330

#### Changes proposed in this Pull Request

* Fix violations of @typescript-eslint/ban-types

Disclaimer: TypeScript is not my strongest skill by any means, I'm not 100% sure of _any_ changes in this PR. I changed the types that seems ok and didn't throw any typing error in VSCode. I'd really appreciate a review, @jsnajdr, @sarayourfriend, @sirreal

### CODE REVIEW

File: client/components/promo-section/promo-card/cta.tsx [hunk 1]
```diff
@@ -119,7 +119,7 @@ const PromoCardCta: FunctionComponent< Props & ConnectedProps > = ( {
 	);
 };
 
-export default connect< ConnectedProps, {}, Props >( ( state, { cta } ) => {
+export default connect< ConnectedProps, unknown, Props >( ( state, { cta } ) => {
```
Comment (scinos) [inline client/components/promo-section/promo-card/cta.tsx:122]: More info https://github.com/typescript-eslint/typescript-eslint/issues/2063

File: client/lib/jetpack/use-track-callback.ts [hunk 2]
```diff
@@ -6,7 +6,11 @@ import getSelectedSiteId from 'calypso/state/ui/selectors/get-selected-site-id';
 // eslint-disable-next-line @typescript-eslint/no-empty-function
 const noop = () => {};
 
-const useTrackCallback = ( callback: Function = noop, eventName: string, eventProps = {} ) => {
+const useTrackCallback = (
+	callback: ( ...args: any[] ) => void = noop,
```
Comment (sirreal) [inline client/lib/jetpack/use-track-callback.ts]: Do we need any or can we prefer the safer alternative `unknown`? 
```suggestion
	callback: ( ...args: unknown[] ) => void = noop,
```

File: client/lib/jetpack/wpcom-upsell-controller.tsx [hunk 3]
```diff
@@ -5,8 +5,10 @@ import { isJetpackSite } from 'calypso/state/sites/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import type { Context } from './types';
 
-export default function upsellSwitch( UpsellComponent: typeof React.Component ): Function {
-	return ( context: Context, next: Function ) => {
+export default function upsellSwitch(
+	UpsellComponent: typeof React.Component
```
Comment (sirreal) [inline client/lib/jetpack/wpcom-upsell-controller.tsx]: You didn't touch this bit, but I believe `typeof` is redundant here:

```suggestion
	UpsellComponent: React.Component
```

File: client/my-sites/plans/jetpack-plans/plan-renewal-message/index.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { useTranslate } from 'i18n-calypso';
 import React, { FunctionComponent } from 'react';
 
-type Props = {};
+type Props = Record< never, never >;
 
 const PlanRenewalMessage: FunctionComponent< Props > = () => {
```
Comment (sirreal) [inline client/my-sites/plans/jetpack-plans/plan-renewal-message/index.tsx]: We shouldn't need to make empty props:

```suggestion
const PlanRenewalMessage: FunctionComponent = () => {
```

Or even:

```suggestion
const PlanRenewalMessage = (): JSX.Element => {
```

**Review by sirreal — APPROVED** [review]
I scanned and left some notes. This seems reasonable, although I haven't tested or inspected in detail.

**Review by sirbrillig — APPROVED** [review]
Approving the changes to the `composite-checkout` package. There don't appear to be any type errors.

**Pull request merged**